Plugins/NativeEngine/Source/VertexBuffer.cpp

```diff
@@ -75,7 +75,22 @@ namespace Babylon
         else
         {
             // Buffer hasn't been finalized yet, all that's necessary is to swap out the bytes.
-            m_bytes = {bytes.data(), bytes.data() + bytes.size()};
+            if (m_bytes && !(*m_bytes).empty())
+            {
+                // update a portion of the vertex buffer bytes
+                memcpy((*m_bytes).data() + byteOffset, bytes.data(), bytes.size());
+            }
+            else
+            {
+                if (byteOffset != 0)
+                {
+                    throw Napi::Error::New(env, "Cannot update a vertex buffer that has no data yet.");
```

Comment:
This doesn't seem right. We should be able to store what needs to be updated and then update it later?

Reply:
No, it's a safe guard that should never happen: You can't update a VB that has not been filled before. But, if ever, then you'll get an exception.